tidy3d/plugins/microwave/path_integrals.py

```diff
@@ -0,0 +1,299 @@
+"""Helper classes for performing path integrals with fields on the Yee grid"""
+
+from __future__ import annotations
+
+import pydantic.v1 as pd
+import numpy as np
+from typing import Tuple
+from ...components.data.dataset import FieldDataset, FieldTimeDataset, ModeSolverDataset
+from ...components.data.data_array import AbstractSpatialDataArray
+from ...components.base import cached_property
+from ...components.types import Axis, Direction
+from ...components.geometry.base import Box
+from ...components.validators import assert_line, assert_plane
+from ...exceptions import DataError
+
+
+class AbstractAxesRH:
+    """Represents an axis-aligned right-handed coordinate system with one axis preferred."""
+
+    @cached_property
+    def main_axis(self) -> Axis:
```

Comment:
make this an `@abc.abstractmethod` too, to enforce that subclasses implement this method.

Then, the `raise NotImplementedError` is not needed,